### PR TITLE
multiSig Fund-transfer testing - add new test-cases

### PR DESCRIPTION
**Pull request by githubckgoh1439**
1. add new test-cases - for MultiSig Fund Transfer.
2. update codec under MultiSig,

### CODE REVIEW

**Comment by githubckgoh1439** [comment]
Hi, Mostafa

Need your comments base on these kind of scenarios:

For those cases base on the 'Delete MultiSig Tx' with those flags which as I expected, 
but with Error when I run. Please help to take look on it.

**Comment by githubckgoh1439** [comment]
Hi, YK

Please comment on this.

We may need to move this checking from '/x/auth/handler.go' to 'app/validation.go'
==> eg. [Delete MultiSig Tx - Error, due to 'Pending tx is not found' which ID :]

where 

Filename : /x/auth/handler.go
[

if pendingTx == nil {
		return sdkTypes.ErrUnknownRequest("Pending tx is not found.").Result()
	}

]

**Comment by yenkhoon** [comment]
only one checking? how bout the others?

**Comment by githubckgoh1439** [comment]
Hi, YK

This I add-in :

cdc.RegisterConcrete(MsgDeleteMultiSigTx{}, "mxw/msgDeleteMultiSigTx", nil)



Regards,

Goh

**Comment by githubckgoh1439** [comment]
Hi, YK

For time-being only that, one issue I already comment in the pull request,
base on the 'Delete MultiSig Tx' as I expected, but run with Error at the
end. Need Boss mostafa to comment and reply.



Regards,
Goh

**Comment by yenkhoon** [comment]
the rest is a very good job.

**Comment by b00f** [comment]
1- Please don't use Tab for indenting. This will make your code be different in other computers. You can force your vscode to always use space instead of tab. Open preferences and search for tab, the enable **Insert Space** feature.

2- Why some tests are commented? like `case-1.1-Delete MultiSig Tx`

3- Transfer ownership to non-kyc owenr. Should it work?

**Comment by githubckgoh1439** [comment]
1- Please don't use Tab for indenting. This will make your code be
different in other computers. You can force your vscode to always use space
instead of tab. Open preferences and search for tab, the enable *Insert
Space* feature.

Goh : OK will use this common-practice.

2- Why some tests are commented? like case-1.1-Delete MultiSig Tx

Goh : due to getting this Error Message of 'Unable to find internal
transaction: '. Need your comments on this. After signed, still allow to
delete the Internal-Tx ?

3- Transfer ownership to non-kyc owenr. Should it work?

Goh : Not in the current test-cases. Will add-in this. It should throw
Error Message instead of proceed.



Regards,

Goh

**Comment by b00f** [comment]
> I had updated those 'Delete MultiSig Tx' in the latest commit today, but still have problems in Case-1.1, Case-2.

Please give me the line number, So many Case-1.1 we have.

**Comment by b00f** [comment]
> 'Unable to find internal transaction'

This error means you are ging to delete an internal transaction which 1) not exist, or 2) it gt broadcast before, or) TX_ID is not correct.

File: x/auth/codec.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ func RegisterCodec(cdc *codec.Codec) {
 	cdc.RegisterConcrete(MsgCreateMultiSigTx{}, "mxw/msgCreateMultiSigTx", nil)
 
 	cdc.RegisterConcrete(MsgSignMultiSigTx{}, "mxw/msgSignMultiSigTx", nil)
+	cdc.RegisterConcrete(MsgDeleteMultiSigTx{}, "mxw/msgDeleteMultiSigTx", nil)
```
[the remark this answers is not in the document]
Comment (b00f) [reply x/auth/codec.go]: Looks you forgot to add this codec. But please double check it.

File: app/validation.go [hunk 2]
```diff
@@ -498,6 +498,10 @@ func (app *mxwApp) validateMsg(ctx sdkTypes.Context, msg sdkTypes.Msg) sdkTypes.
 		if !groupAcc.GetMultiSig().IsOwner(msg.Sender) {
 			return sdkTypes.ErrUnknownRequest("Only group account owner can remove pending tx.")
 		}
+		pendingTx := groupAcc.GetMultiSig().GetPendingTx(msg.TxID)
```
Comment (b00f) [inline app/validation.go]: Good found! Thanks.

**Pull request merged**